File: bindings/java/src/main/java/io/github/endeavoury/ratatoskr/DnsRecordType.java
package io.github.endeavoury.ratatoskr;

import java.util.Arrays;
import java.util.Optional;

/** DNS resource-record types understood by the initial Ratatoskr ABI. */
public enum DnsRecordType {
    A(1), NS(2), MD(3), MF(4), CNAME(5), SOA(6), MB(7), MG(8), MR(9),
    NULL(10), WKS(11), PTR(12), HINFO(13), MINFO(14), MX(15), TXT(16),
    AAAA(28), SRV(33), NAPTR(35), CAA(257);

    private final int code;

    DnsRecordType(int code) {
        this.code = code;
    }

    /** Returns the unsigned DNS type code. */
    public int code() {
        return code;
    }

    /** Returns the known type for a numeric code, or an empty value for an unknown RR. */
    public static Optional<DnsRecordType> fromCode(int code) {
        return Arrays.stream(values()).filter(value -> value.code == code).findFirst();
    }
}

File: bindings/php/src/DnsRecordType.php
<?php
declare(strict_types=1);

namespace Ratatoskr;

enum DnsRecordType: int
{
    case A = 1; case NS = 2; case CNAME = 5; case SOA = 6; case PTR = 12;
    case MX = 15; case TXT = 16; case AAAA = 28; case SRV = 33;
    case NAPTR = 35; case CAA = 257;
}
